src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs

```diff
@@ -161,10 +163,42 @@ void BindIndicatorItems()
 				}
 			});
 
-			BindableLayout.SetItemsSource(this, _indicatorView.ItemsSource);
+			// Get the filtered items source based on MaximumVisible
+			var itemsSource = GetFilteredItemsSource();
+			BindableLayout.SetItemsSource(this, itemsSource);
+
 			BindableLayout.SetItemTemplate(this, indicatorTemplate);
 		}
 
+		IEnumerable? GetFilteredItemsSource()
+		{
+			if (_indicatorView.ItemsSource is null || _indicatorView.MaximumVisible <= 0)
+			{
+				return null;
+			}
+
+			if (_indicatorView.ItemsSource is IList items)
+			{
+				if (items.Count <= _indicatorView.MaximumVisible)
+				{
+					return items;
+				}
+
+				var filteredItems = new List<object>();
```

Comment:
Using the `MaximumVisible` property here can create a pre-sized List:

`var filteredItems = new List<object>(_indicatorView.MaximumVisible);`